### ISSUE TITLE
Lint/UselessAccessModifier doesn't recognize conditionally defined methods

### ISSUE THREAD

**Issue by bquorning** [bbatsov/rubocop#3049]
When a class or module uses `if` to conditionally define a private method, the `Lint/UselessAccessModifier` cop seems to thing that the `private` access modifier is useless.

## Expected behavior

No offenses.

## Actual behavior

One offense:

```
test.rb:8:3: W: Useless private access modifier.
  private
  ^^^^^^^
```

## Steps to reproduce the problem

Run `rubocop` on this (contrived) test.rb file:
```rb
# encoding: utf-8
# frozen_string_literal: true
class Foo
  def initialize
    puts value
  end

  private

  if defined?(Bar)
    def value
      true
    end
  else
    def value
      false
    end
  end
end
```

## RuboCop version

```
» bin/rubocop -V
0.39.0 (using Parser 2.3.0.7, running on ruby 2.3.0 x86_64-darwin15)
» git log -1 --pretty=format:'%H'
3564db025d475397ab2a0ebb9d584a114421c198
```
